File: output/ChatGPT/avatar/Python/C++/codeforces_148_B.cpp
#include <iostream>
using namespace std;

int main() {
    int princess, dragon, t, f, c;
    cin >> princess >> dragon >> t >> f >> c;
    
    if (dragon <= princess) {
        cout << 0 << endl;
        exit(0);
    }
    
    int result = 0;
    int position = t * princess;
    position += princess * position / (dragon - princess);
    
    while (position < c) {
        result += 1;
        position += princess * position / dragon;
        position += princess * f;
        position += princess * position / (dragon - princess);
    }
    
    cout << result << endl;
    
    return 0;
}

File: output/CodeLlama/avatar/Python/Java/codeforces_148_B.java
import java.util.Scanner;
public class codeforces_148_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int princess = sc.nextInt();
        int dragon = sc.nextInt();
        int t = sc.nextInt();
        int f = sc.nextInt();
        int c = sc.nextInt();
        if (dragon <= princess) {
            System.out.println(0);
            return;
        }
        int result = 0;
        int position = t * princess;
        position += princess * position / (dragon - princess);
        while (position < c) {
            result++;
            position += princess * position / dragon;
            position += princess * f;
            position += princess * position / (dragon - princess);
        }
        System.out.println(result);
    }
}